src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -85,6 +85,58 @@ class java_bytecode_convert_classt:public messaget
   static void add_array_types(symbol_tablet &symbol_table);
 };
 
+/// Auxiliary function to extract the generic superclass reference from the
+/// class signature. If the superclass is not generic, it returns an empty
+/// string.
+/// \param signature Signature of the class
+/// \return Reference of the generic superclass, or empty if the superclass
+/// is not generic
+static std::string
+extract_generic_superclass_reference(const std::string &signature)
+{
+  // skip the (potential) list of generic parameters at the beginning of the
+  // signature
+  const size_t start = signature.front() == '<'
+                         ? find_closing_delimiter(signature, 0, '<', '>') + 1
+                         : 0;
+
+  // extract the superclass reference
+  const std::string superclass_ref = signature.substr(
+    start,
+    find_closing_semi_colon_for_reference_type(signature, start) - start + 1);
+
+  // if the superclass is generic then the reference is of form
+  // Lsuperclass-name<generic-types>;
+  if(superclass_ref.substr(superclass_ref.length() - 2) == ">;")
+    return superclass_ref;
+  else
+    return "";
+}
+
+/// Auxiliary function to extract the generic interface reference of an
+/// interface with the specified name from the class signature. If the
+/// interface is not generic, it returns an empty string.
+/// \param signature Signature of the class
+/// \param interface_name The interface name
+/// \return Reference of the generic interface, or empty if the interface
+/// is not generic
+static std::string extract_generic_interface_reference(
+  const std::string &signature,
+  const std::string &interface_name)
+{
+  if(signature.find("L" + interface_name + "<") != std::string::npos)
+  {
+    const size_t start = signature.find("L" + interface_name + "<");
```

Comment:
Would it be better to move this `start` variable and its expression outside the `if()` statement and only do the `find` once?